package/src/hooks/useQueryReminders.ts

```diff
@@ -0,0 +1,150 @@
+          newData.push(reminder);
+        } else {
+          if (!deletedOrUpdatedRemindersCache.current[reminder.message_id]) {
+            newData.push(reminder);
+          }
+        }
+      });
+      return newData;
+    });
+  }, [items, client.reminders.paginator.filters]);
+
+  useEffect(() => {
+    const handleReminderDeleted = (event: Event) => {
+      if (!event.reminder?.message_id) {
+        return;
+      }
+      deletedOrUpdatedRemindersCache.current[event.reminder.message_id] = event.reminder;
+      setData((prevData) => {
+        return prevData.filter((item) => item.message_id !== event.reminder?.message_id);
+      });
+    };
+
+    const handleReminderCreated = (event: Event) => {
+      setData((prevData) => {
+        if (!event.reminder) {
+          return prevData;
+        }
+        const updatedData = [...prevData, event.reminder];
+        return sortRemindersByDate(updatedData);
+      });
+    };
+
+    const handleReminderUpdated = (event: Event) => {
+      const { reminder } = event;
+      if (!reminder || showAllReminders(client.reminders.paginator.filters)) {
+        return; // No update needed if reminder is undefined or filters is empty
+      }
+      deletedOrUpdatedRemindersCache.current[reminder.message_id] = reminder;
+      setData((prevData) => {
+        const existingReminder = prevData.find((item) => item.message_id === reminder?.message_id);
+        if (!existingReminder) {
+          return prevData; // No update needed if reminder not found
+        }
+
+        if (existingReminder.remind_at && !event.reminder?.remind_at) {
+          return prevData.filter((item) => item.message_id !== event.reminder?.message_id);
+        }
+        if (!existingReminder.remind_at && event.reminder?.remind_at) {
+          return prevData.filter((item) => item.message_id !== event.reminder?.message_id);
+        }
+        if (isReminderOverdue(existingReminder) && !isReminderOverdue(event.reminder)) {
+          return prevData.filter((item) => item.message_id !== event.reminder?.message_id);
+        }
+        if (isReminderUpcoming(existingReminder) && !isReminderUpcoming(event.reminder)) {
+          return prevData.filter((item) => item.message_id !== event.reminder?.message_id);
+        }
+
+        return prevData;
+      });
+    };
+
+    const listeners = [
+      client.on('reminder.created', handleReminderCreated),
+      client.on('reminder.deleted', handleReminderDeleted),
+      client.on('reminder.updated', handleReminderUpdated),
+    ];
+
+    return () => {
+      listeners.forEach((l) => l.unsubscribe());
+    };
+  }, [client]);
+
+  const onEndReached = useCallback(async () => {
+    await client.reminders.queryNextReminders();
+  }, [client.reminders]);
+
+  const onRefresh = useCallback(async () => {
+    await client.reminders.queryNextReminders();
+  }, [client.reminders]);
+
+  return {
+    data,
+    isLoading,
+    onEndReached,
+    onRefresh,
```

Comment:
`onRefresh` means nothing in scope of this hook - they should be named this in the sample app maybe but not here as well